lib/gears/filesystem.lua

```diff
@@ -0,0 +1,48 @@
+---------------------------------------------------------------------------
+--- Filesystem module for gears
+--
+-- @module gears.filesystem
+---------------------------------------------------------------------------
+
+-- Grab environment we need
+local Gio = require("lgi").Gio
+
+local filesystem = {}
+
+--- Create a directory
+-- @tparam string dir The directory.
+-- @return mkdir return code
+function filesystem.mkdir(dir)
+    return os.execute("mkdir -p " .. dir)
```

Comment:
Could you replace this with:
```lua
local gfile = Gio.File.new_for_path(filename)
gfile:make_directory_with_parents()
```
And, actually, some return values might be nice.
The following is untested, but might work (API documentation is missing):
```lua
local gfile = Gio.File.new_for_path(filename)
local success, err = gfile:make_directory_with_parents()
if success then
    return true
end
if err.domain = Gio.IOErrorEnum and err.code == "EXISTS" then
    -- Directory already exists, let this count as success
    return true
end
return false
```